### ISSUE TITLE
RPC calls for egihash info

### ISSUE THREAD

**Issue by RyanLucchese** [energicryptocurrency/energi#154]
We need some new RPC calls to return info about egihash state.

* `getepoch` - should return the current epoch number `(chainheight / egihash::constants::EPOCH_LENGTH)`
* `getseedhash ( n )` - should return the hex encoded seedhash for specified epoch `n` or the current epoch if `n` is not specified
* `getdagsize ( n )` - should return the size of the DAG in bytes for specified epoch `n` or the current epoch if `n` is not specified
* `getdagcachesize ( n )` - should return the size of the DAG cache in bytes for specified epoch `n` or the current epoch if `n` is not specified
* `getdag ( n )` - should return a JSON object specifying DAG information:
```
{
  "epoch": n,
  "seedhash": "0000000000000000000000000",
  "size": 9999999999999999,
  "cache_size": 999999999,
}
```
* `getcache ( n )` - should return a JSON object specifying DAG cache information:
```
{
  "epoch": n,
  "seedhash": "0000000000000000000000000",
  "size": 999999999,
}
```
* `getactivedag` - should return a JSON list specifying loaded DAGs, same output as `getdag n` in an array for values of n that are loaded

**Comment by RyanLucchese** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [RPC calls for egihash info](https://trello.com/c/cz4XelgI/98-rpc-calls-for-egihash-info)

**Comment by RyanLucchese** [comment]
Fixed by #189
